eth/ethconfig/gen_config.go

```diff
@@ -3,46 +3,92 @@
 package ethconfig
 
 import (
+	"math/big"
 	"time"
 
 	"github.com/c2h5oh/datasize"
 	"github.com/ledgerwatch/erigon-lib/chain"
-	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon-lib/common/datadir"
+	"github.com/ledgerwatch/erigon-lib/downloader/downloadercfg"
+	"github.com/ledgerwatch/erigon-lib/txpool/txpoolcfg"
+	"github.com/ledgerwatch/erigon/cl/beacon/beacon_router_configuration"
+	"github.com/ledgerwatch/erigon/cl/clparams"
 	"github.com/ledgerwatch/erigon/consensus/ethash/ethashcfg"
 	"github.com/ledgerwatch/erigon/core/types"
 	"github.com/ledgerwatch/erigon/eth/gasprice/gaspricecfg"
-
 	"github.com/ledgerwatch/erigon/ethdb/prune"
 	"github.com/ledgerwatch/erigon/params"
 )
 
 // MarshalTOML marshals as TOML.
 func (c Config) MarshalTOML() (interface{}, error) {
 	type Config struct {
+		Sync                           Sync
 		Genesis                        *types.Genesis `toml:",omitempty"`
 		NetworkID                      uint64
 		EthDiscoveryURLs               []string
 		Prune                          prune.Mode
 		BatchSize                      datasize.ByteSize
 		ImportMode                     bool
-		BadBlockHash                   libcommon.Hash
+		BadBlockHash                   common.Hash
 		Snapshot                       BlocksFreezing
-		BlockDownloaderWindow          int
+		Downloader                     *downloadercfg.Cfg
+		BeaconRouter                   beacon_router_configuration.RouterConfiguration
+		CaplinConfig                   clparams.CaplinConfig
+		Dirs                           datadir.Dirs
 		ExternalSnapshotDownloaderAddr string
-		Whitelist                      map[uint64]libcommon.Hash `toml:"-"`
+		Whitelist                      map[uint64]common.Hash `toml:"-"`
 		Miner                          params.MiningConfig
 		Ethash                         ethashcfg.Config
 		Clique                         params.ConsensusSnapshotConfig
 		Aura                           chain.AuRaConfig
-		TxPool                         DeprecatedTxPoolConfig
+		DeprecatedTxPool               DeprecatedTxPoolConfig
+		TxPool                         txpoolcfg.Config
 		GPO                            gaspricecfg.Config
 		RPCGasCap                      uint64  `toml:",omitempty"`
 		RPCTxFeeCap                    float64 `toml:",omitempty"`
 		StateStream                    bool
-		BodyDownloadTimeoutSeconds     int
-		SyncLoopThrottle               time.Duration
+		HistoryV3                      bool
```

Comment:
Why are all of these more general Erigon parameters changing in this PR?

Reply:
We haven't run the `go:generate gencodec -dir . -type Config -formats toml -out gen_config.go` command in a while, so some variables have been deprecated, and some new variables have been added.